spec/active_delivery/base_spec.rb

```diff
@@ -229,21 +229,38 @@ def do_anything
           before_notify :ensure_id_positive
           before_notify :ensure_duck_present, on: :quack
 
+          after_notify :launch_fireworks, on: :quack, only: %i[do_something]
+          after_notify :kill_duck, except: %i[do_something]
+
           def ensure_id_positive
             params[:id] > 0
           end
 
           def ensure_duck_present
             params[:duck].present?
           end
+
+          def launch_fireworks
+          end
+
+          def kill_duck
+          end
         end
       )
     end
 
     specify "when callbacks pass" do
+      expect_any_instance_of(delivery_class).to receive(:launch_fireworks)
```

Comment:
Let's test behaviour not implementation, i.e. not use mocks/stubs (especially `any_instance`-kind) but use the existing testing mechanism instead (`call << "smth"`).